src/realm/collection_parent.hpp

```diff
@@ -65,7 +65,53 @@ enum class UpdateStatus {
 };
 
 
-using PathElement = Mixed;
+class PathElement : private Mixed {
```

Comment:
This class is going to simplify a little bit the code. Probably worth to expose in the C-API, however we still have the problem that we need to return the actual type, either list or dictionary. So I guess the API still needs to look like:

realm_list_t* list_add_list(size_t); //it can only be integer
realm_dictionary_t* list_add_dictionary(size_t);  
// same for dictionary.